### DISCUSSION TITLE
Allow hosting of custom backend at /api

### DISCUSSION THREAD

**Discussion by bdevwebsolutions** [vercel/next.js#47400] [Help]
### Summary

Currently we are working on a project using CRA and a custom backend (.NET), we host the application with Docker, routing it so that all calls to the HOST url followed with /api are calls to the backend (as how the nextjs library handles api routes).

so for example:
- https://www.app.be/ => React app
- https://www.app.be/api/products => Api call to .NET backend

As i would like to transition from cra to nextjs because of the features included i would like to know if it's possible to disable the /api functionality in Nextjs and just keep our deployment as is using our .NET backend trough /api

**Comment by icyJoseph** [comment]
Hi,

I think you can use rewrites, as detailed by the incremental adoption guide: https://nextjs.org/docs/migrating/incremental-adoption#rewrites

So for example, say I've got:

```js
// next.config.js
  async rewrites() {
    return {
      beforeFiles: [
        {
          source: "/api/:path*",
          destination: `https://pokeapi.co/api/v2/:path*`,
        },
      ],
    };
  },
```

Then any request arriving to my Next.js server's `/api/something-something` is passed to the Pokemon API.

![Screenshot 2023-03-22 at 16 18 02](https://user-images.githubusercontent.com/21013447/226951671-a80a1278-d678-4000-a877-ca1ccba4d8cd.png)

Try it out, let me know how it goes, I can help out if something doesn't go swell.

**Reply by bdevwebsolutions** [reply]
Surely this would be ideal if our backend would be hosted at a different url as our frontend but that's not the case. our source and destination are hosted at the same url with our backend occupying the /api routes
